### PR TITLE
Configure devise for project

### PR DESCRIPTION
**Pull request by martinjaimem**
Configure devise for the project.
 - Add devise endpoints: sign in, sign up, reset pass.

### CODE REVIEW

**Comment by GuilleLeopold** [comment]
Let me know when it's ready for review. In the meanwhile I edited the name and I added a label to mark it as `wip`

**Comment by martinjaimem** [comment]
@GuilleLeopold It's ready for review

**Comment by diebarral** [comment]
Please add a description to the PR

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,6 +1,23 @@
+# app/controllers/application_controller.rb
 class ApplicationController < ActionController::Base
+  include DeviseTokenAuth::Concerns::SetUserByToken
+
+  # Prevent CSRF attacks by raising an exception.
+  # For APIs, you may want to use :null_session instead.
```
Comment (diebarral) [inline app/controllers/application_controller.rb]: Remove commented lines

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,6 +1,23 @@
...
   # GET /status
   def status
-    render json: { 'online': true }
+    render json: { online: true }
+  end
+
+  protected
+
+  def configure_permitted_parameters
+    # Permit the `subscribe_newsletter` parameter along with the other
```
Comment (diebarral) [inline app/controllers/application_controller.rb]: Also remove commented lines here

File: app/controllers/api/v1/users_controller.rb [hunk 3]
```diff
@@ -1,11 +1,10 @@
-# frozen_string_literal: true
-
 module Api
   module V1
     class UsersController < ApplicationController
+      include DeviseTokenAuth::Concerns::SetUserByToken
```
Comment (GuilleLeopold) [inline app/controllers/api/v1/users_controller.rb]: move this to the application controller

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -1,6 +1,23 @@
+# app/controllers/application_controller.rb
 class ApplicationController < ActionController::Base
+  include DeviseTokenAuth::Concerns::SetUserByToken
```
Comment (GuilleLeopold) [inline app/controllers/application_controller.rb]: same here

File: app/models/user.rb [hunk 5]
```diff
@@ -27,12 +27,24 @@
 #  updated_at             :datetime         not null
 #
 
-
 class User < ApplicationRecord
-  validates :gender, presence: true
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
```
Comment (GuilleLeopold) [inline app/models/user.rb]: Remove this lines

File: app/models/user.rb [hunk 6]
```diff
@@ -27,12 +27,24 @@
...
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+  include DeviseTokenAuth::Concerns::User
+
   enum gender: { male: 0, female: 1, other: 2 }
 
+  before_validation :init_uid
```
Comment (GuilleLeopold) [inline app/models/user.rb]: callbacks go after validations
Check here: https://github.com/rootstrap/tech-guides/blob/master/ruby/rails.md#macro-style-methods

Comment (GuilleLeopold) [reply app/models/user.rb]: You are doing this in a before validations, that means that this code will always run even if you are updating the record, not just creating. Check on how to run this callback only for creation: `on: :create` (check syntax, not sure)

Comment (GuilleLeopold) [reply app/models/user.rb]: Anyway, are you sure that devise doesn't already do this?

Comment (martinjaimem) [reply app/models/user.rb]: You are right it, it isn't necessary 👍

**Comment by diebarral** [comment]
As a comment, try to name the branches according to the organizations guidelines as can be found [here](https://github.com/rootstrap/tech-guides/tree/master/git#branches)

File: app/controllers/api/v1/users_controller.rb [hunk 7]
```diff
@@ -1,11 +1,9 @@
-# frozen_string_literal: true
-
 module Api
   module V1
     class UsersController < ApplicationController
+      before_action :authenticate_user!
       helper_method :user
```
Comment (jdrosales17) [inline app/controllers/api/v1/users_controller.rb]: Put the helper method after the `before_action` calls

File: app/controllers/api/v1/users_controller.rb [hunk 8]
```diff
@@ -14,15 +12,6 @@ def index
...
-        @user = User.new(user_params)
-        if @user.save
-          render :show
-        else
-          render json: @user.errors, status: :unprocessable_entity
-        end
-      end
-
       def update
         @user.update!(user_params)
```
Comment (jdrosales17) [inline app/controllers/api/v1/users_controller.rb]: You can do just `user.update!(user_params)` here

**Pull request merged**